File: samples/NdkGdbSample/app/src/main/java/com/android/developer/ndkgdbsample/MainActivity.java
package com.android.developer.ndkgdbsample;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        TextView tv = new TextView(this);
        setContentView(tv);
        tv.setText(getHelloString());
    }

    private native String getHelloString();

    static {
        System.loadLibrary("app");
    }
}

File: samples/NdkGdbSample/app/src/main/java/com/android/developer/ndkgdbsample/MainActivity.kt
package com.android.developer.ndkgdbsample

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val tv = findViewById<View>(R.id.hello_textview) as TextView
        tv.text = getHelloString()
    }

    private external fun getHelloString(): String

    companion object {
        init {
            System.loadLibrary("app")
        }
    }
}
